### PR TITLE
Install vagrant from the upstream binary instead of from openSUSE

### PR DESCRIPTION
**Pull request by dcermak**
This is necessary for the vagrant tests to continue to run as vagrant will be dropped due to the license change.

- Related ticket: na
- Needles: na
- Verification run: ~~https://openqa.opensuse.org/tests/3813017~~ ~~https://openqa.opensuse.org/tests/3816011~~ ~~https://openqa.opensuse.org/tests/3821702~~ ~~https://openqa.opensuse.org/tests/3823877~~ ~~https://openqa.opensuse.org/tests/3824273~~ https://openqa.opensuse.org/tests/3824410

### CODE REVIEW

**Comment by dcermak** [comment]
@ggardet I had to drop arm testing as we're dropping vagrant and vagrant-libvirt isn't playing nice with the upstream binary

**Comment by ggardet** [comment]
Could you update https://github.com/os-autoinst/opensuse-jobgroups/blob/master/job_groups/opensuse_tumbleweed_aarch64.yaml to drop vagrant, please?

For the record, the upstream issue for upstream arm64 binaries: https://github.com/hashicorp/vagrant/issues/12548

**Comment by dcermak** [comment]
> Could you update https://github.com/os-autoinst/opensuse-jobgroups/blob/master/job_groups/opensuse_tumbleweed_aarch64.yaml to drop vagrant, please?

https://github.com/os-autoinst/opensuse-jobgroups/pull/403

**Review by ggardet — APPROVED** [review]

File: tests/virtualization/vagrant/boxes/tumbleweed.pm [hunk 1]
```diff
@@ -46,12 +45,13 @@ sub run() {
...
 
-    my %boxes = (
-        # Tumbleweed.x86_64-1.0-{libvirt|virtualbox}-Snapshot20190704.vagrant.{libvirt|virtualbox}.box
-        libvirt => "$version.$arch-1.0-libvirt$arch_ext-Snapshot$build.vagrant.libvirt.box",
-    );
-    # virtualbox is supported only on x86_64
-    %boxes = (%boxes, virtualbox => "$version.$arch-1.0-virtualbox-Snapshot$build.vagrant.virtualbox.box") if $is_virtualbox_applicable;
+    my %boxes = ();
+    if ($is_virtualbox_applicable) {
+        my %boxes = (
```
Comment (Vogtinator) [inline tests/virtualization/vagrant/boxes/tumbleweed.pm:50]: This doesn't assign but creates a new variable

**Review by Vogtinator — CHANGES_REQUESTED** [review]
The VR doesn't test anything, it only sets up and exits

File: tests/virtualization/vagrant/boxes/tumbleweed.pm (hunk 1 above)
Comment (dcermak) [reply tests/virtualization/vagrant/boxes/tumbleweed.pm]: Should be fixed now

**Comment by dcermak** [comment]
> The VR doesn't test anything, it only sets up and exits

Should be fixed now

**Comment by Vogtinator** [comment]
> > The VR doesn't test anything, it only sets up and exits
> 
> Should be fixed now

That is fixed but now every single vagrant command fails with 127 which the test ignores...

**Comment by dcermak** [comment]
@Vogtinator I've switched to use the vagrant rpm from Hashicorp, which now actually runs. Although I must confess that I'm wondering why there's a kernel panic in the output: https://openqa.opensuse.org/tests/3821702#step/add_box_virtualbox/41 (but the whole thing appears to work…)

File: lib/vagrant.pm [hunk 2]
```diff
@@ -7,22 +7,33 @@ use utils;
...
+    # jq required for parsing the json
+    zypper_call('in jq');
+    assert_script_run('curl -O https://checkpoint-api.hashicorp.com/v1/check/vagrant');
+
+    my $vagrant_version = script_output('jq -r ".current_version" < vagrant');
+    my $download_url = script_output('jq -r ".current_download_url" < vagrant');
+
+    # https://releases.hashicorp.com/vagrant/2.4.0/vagrant-2.4.0-1.x86_64.rpm
+    my $vagrant_rpm = 'vagrant-' . $vagrant_version . '-1.x86_64.rpm';
+    assert_script_run('curl -O "' . $download_url . $vagrant_rpm . '"');
```
Comment (Vogtinator) [inline lib/vagrant.pm:25]: Both RPM and zypper can work with http URLs directly

**Review by Vogtinator — CHANGES_REQUESTED** [review]
Please fix `run_vagrant_cmd` to not ignore failures. Currently the test is useless as it'll always pass

File: lib/vagrant.pm [hunk 3]
```diff
@@ -33,9 +43,8 @@ sub run_vagrant_cmd {
 
     my $logfile = 'vagrant_cmd.log';
     local $@;
-    my $ret = eval {
-        return script_run("VAGRANT_LOG=DEBUG vagrant $cmd 2> $logfile", %args);
-    };
+    my $ret = script_run("VAGRANT_LOG=DEBUG vagrant $cmd 2> $logfile", %args);
+
     return undef if $ret == 0;
     if (!$ret) {
```
Comment (Vogtinator) [inline lib/vagrant.pm:49]: The bug is probably here

File: lib/vagrant.pm [hunk 4]
```diff
@@ -33,9 +43,8 @@ sub run_vagrant_cmd {
 
     my $logfile = 'vagrant_cmd.log';
     local $@;
```
Comment (Vogtinator) [inline lib/vagrant.pm:45]: I don't even know what this does

File: lib/vagrant.pm (hunk 3 above)
Comment (dcermak) [reply lib/vagrant.pm]: the `if (!$ret)` was not working for reasons. It's working now.

File: lib/vagrant.pm (hunk 4 above)
Comment (Vogtinator) [reply lib/vagrant.pm:45]: Apparently this was for the `eval` so can be removed now

File: lib/vagrant.pm [hunk 5]
```diff
@@ -33,16 +43,14 @@ sub run_vagrant_cmd {
...
-    if (!$ret) {
-        upload_logs($logfile);
 
-        if ($@) {
-            die $@;
-        }
-        die "'vagrant $cmd' failed with $ret";
+    upload_logs($logfile);
+
+    if ($@) {
```
Comment (Vogtinator) [inline lib/vagrant.pm:52]: The eval is gone so this won't work anymore either

File: lib/vagrant.pm [hunk 6]
```diff
@@ -33,16 +43,14 @@ sub run_vagrant_cmd {
 
     my $logfile = 'vagrant_cmd.log';
     local $@;
-    my $ret = eval {
-        return script_run("VAGRANT_LOG=DEBUG vagrant $cmd 2> $logfile", %args);
-    };
+    my $ret = script_run("VAGRANT_LOG=DEBUG vagrant $cmd 2> $logfile", %args);
```
Comment (Vogtinator) [inline lib/vagrant.pm:46]: This is essentially just `assert_script_run` now

File: lib/vagrant.pm (hunk 3 above)
Comment (Vogtinator) [reply lib/vagrant.pm]: Because

```
    return undef if $ret == 0;
    if (!$ret) {
    // unreachable!
```

File: lib/vagrant.pm (hunk 6 above)
Comment (dcermak) [reply lib/vagrant.pm:45]: Except that it uploads the full vagrant debug log.

File: lib/vagrant.pm (hunk 5 above)
Comment (dcermak) [reply lib/vagrant.pm]: removed

File: lib/vagrant.pm (hunk 6 above)
Comment (dcermak) [reply lib/vagrant.pm]: done

File: tests/virtualization/vagrant/boxes/tumbleweed.pm [hunk 7]
```diff
@@ -80,4 +80,8 @@ sub run() {
     assert_script_run("popd");
 }
 
+sub post_fail_hook() {
+    upload_logs($vagrant_logfile);
```
Comment (Vogtinator) [inline tests/virtualization/vagrant/boxes/tumbleweed.pm:84]: Missing `$self->SUPER::post_fail_hook;`, also in the other module (before this PR already...)

**Review by Vogtinator — APPROVED** [review]
LGTM, just a minor issue with the post_fail_hooks

File: tests/virtualization/vagrant/boxes/tumbleweed.pm (hunk 7 above)
Comment (Vogtinator) [reply tests/virtualization/vagrant/boxes/tumbleweed.pm:84]: Needs `my ($self) = @_;`

Comment (Vogtinator) [reply tests/virtualization/vagrant/boxes/tumbleweed.pm:84]: Needs something like `my ($self) = @_;`

**Comment by dcermak** [comment]
last VR passed: https://openqa.opensuse.org/tests/3824436

**Pull request merged**